### DISCUSSION TITLE
AP가 제일 낮은 General trash class에 대한 인사이트 공유

### DISCUSSION THREAD

**Discussion by jaehwan-AI** [boostcampaitech2/object-detection-level2-cv-06#143] [Show and tell]
얼핏 보기에도 종류가 많고 일관성이 없는 class인 General trash(이하 일반쓰레기)에 대해서 faster RCNN 모델의 결과로 분석해 본 결과입니다.
모델의 mAP 평가 관점이나 robust한 모델을 만드는 관점에서는 직접적으로 거리가 멀 수 있으나 가장 기본 모델에서 데이터가 잘 학습되지 않는 이유에 대해서 알아보고자 해당 작업을 시작했습니다.

validation set을 inference한 결과를 토대로 분석을 했으며, 분석 중간중간에 주관적인 인사이트도 공유하는 글이니 가벼운 마음으로 읽으시면 될 것 같습니다. (+ 쓰다보니 길어졌네요.. 스크롤 쭉쭉 넘겨주세요 ㅎㅎ)
지금부터 보실 _**### 이미지의 왼쪽은 faster RCNN의 inference 결과 / 오른쪽은 validation set의 Ground Truth**_ 입니다.

## 1. **흰 쓰레기(휴지)를 대부분 paper로 인식 -> 종이와 종이팩, 일반쓰레기의 경계가 모호합니다.**
![image](https://user-images.githubusercontent.com/66052001/137248949-1c4a8767-d540-4c28-82bd-9c096894b689.png)
![image](https://user-images.githubusercontent.com/66052001/137248996-a0596ccd-8cde-4f70-b988-7bef17870fa2.png)
![image](https://user-images.githubusercontent.com/66052001/137249017-260e565c-368c-4959-8e9d-7bebe50de58e.png)
![image](https://user-images.githubusercontent.com/66052001/137249028-436085f0-5e33-4701-88b4-f630644ecf17.png)
![image](https://user-images.githubusercontent.com/66052001/137249276-5882e8f8-51b8-45c6-a9cb-0a6d540bb652.png)
-> 흰색 바탕 + 글씨 + 컬러 등의 헷갈리는 요소가 많습니다.
-> **(주관적인 가설)**관련 augmentation을 통해 하나의 feature에 치우치지 않게 학습을 진행해 볼 수 있을 것 같습니다.

## 2. 담배갑에 있는 비닐을 plastic bag으로 인식 + 추가 인식 데이터 발생
![image](https://user-images.githubusercontent.com/66052001/137249484-9fb96394-cd55-4185-8b1b-ff7668f39963.png)
![image](https://user-images.githubusercontent.com/66052001/137249698-4b02d6aa-b518-4c03-bb35-dbd67e97a3cf.png)
![image](https://user-images.githubusercontent.com/66052001/137249713-f71f3524-ecb7-44e5-9183-e8092fc61153.png)
-> **(주관적인 가설)**오히려 추가 인식한 데이터를 이용하여 재학습할 경우 데이터에 일관성이 생겨서 학습에 혼동을 줄여줄 수 있을 것 같습니다.

## 3. 스티로폼에 붙어있는 테이프
![image](https://user-images.githubusercontent.com/66052001/137249961-ab9451da-058d-4625-8062-118cbed7786e.png)
![image](https://user-images.githubusercontent.com/66052001/137249968-b953c29f-8a56-4500-8eae-7612f1f4669c.png)
-> 박스 전체에 테이프가 붙어 있으면 박스 전체를 일반쓰레기 라벨링을 해버리는 경우도 있습니다.
-> 또한 투명 테이프의 경우 사람의 눈으로도 식별하기 힘듭니다.
-> **해당 task는 포기하는 것이 여러모로 좋을 것 같다는 생각입니다..**

## 4. 장갑의 종류에 따라 달라지는 라벨링
![image](https://user-images.githubusercontent.com/66052001/137250318-35f34530-0034-4669-9d14-72494ed9f764.png)
![image](https://user-images.githubusercontent.com/66052001/137250347-2ae5e31b-9493-48e1-a0f3-0f67d098d224.png)
-> 장갑의 종류에 따라 라벨링이 바뀌는 데 적은 데이터에 모델이 장갑의 종류를 구분할 수 있을까? 라는 의문입니다.
-> 반대로 생각하면 장갑이라는 것을 모양에 중점을 두어서 학습을 한 것도 같다는 생각이 듭니다. 하지만 데이터 전체를 분석한 것이 아니기 때문에 성급히 일반화도 하기 어려울 것 같네요..

## 데이터 re-labeling
데이터를 세분화해서 re-labeling 한다면 깔끔한 학습을 할 수 있을 것 같습니다.
예시) 일반쓰레기(담배갑, 장갑, 휴지 등등) -> 담배갑 / 장갑 / 휴지 등으로 annotation을 세분화
-> **시간을 많이 필요로 해서 대회에는 부적합한 방법**

## 대회 관점에서의 mAP
어차피 mAP는 각 class의 AP의 평균입니다. 이 점을 이용한다면 맞추기 힘든 일반쓰레기를 맞추기 위해 노력하기 보다는 맞추기 쉬울 것으로 예상되는 라벨의 AP를 올리는 것도 빠르게 mAP를 올리는 방법이 될 수 있을 것 같습니다.
하지만 이 방법은 우리의 궁극적인 목표인 robust한 모델을 만드는 것에 반대되는 방법이므로 대회에 한정해서 생각해보면 좋을 것 같아서 공유드립니다.
[게시글 댓글 참고](https://stages.ai/competitions/76/discussion/talk/post/709)

이외에도 마스크를 Clothing으로 분류하는 등의 문제가 많았지만 보면서 떠오른 것만 정리해서 적었습니다.
다음 대회에는 샛길로 가지 않고 도움을 많이 드릴 수 있게 노력하겠습니다! 칠성사이다 화이팅!!

**Comment by ppskj178** [comment]
re-labeling -> 시간을 많이 필요로 해서 대회에는 부적합한 방법 -> 돈을 필요로 해서 회사에서도....

T^T

데이터셋 엉망으로 만든 외주업체를 찾아가는게 맞지않나.....

**Reply by jaehwan-AI** [reply]
일단 외주업체 박살내야죠 ㅋㅋㅋㅋㅋ
하지만 데이터를 수집하는 과정이 어려운 데이터라면 그냥 re-labeling하는 것도 방법이 될 수 있는 것 같아요
cost(re-labeling) < cost(collecting)
즉 데이터를 사용하는 편이 낫다면이라는 조건이 붙을 것 같네요

**Reply by ppskj178** [reply]
먼 미래까지 버틴다면 오히려 좋을지도 라는 생각이드네요 ㅋㅋ
어차피 스트로폼에 테이프 쳐놓은것도 결국 뗴서 분리해야 되는 거니까
인공지능이 엄청 발달한 후에는 저렇게 라벨링한 데이터가 필요할지도....

**Reply by jaehwan-AI** [reply]
그 인공지능 얼른 보고 싶네요 ㅋㅋㅋ

**Comment by JiyouSeo** [comment]
칠성사이다 화이팅!!!
